### PR TITLE
Update BqVisualiser to Angular V12

### PR DESCRIPTION
**Pull request by smeyn**
/tools/BQvisualizer has been updgraded to Angular V 12, as users had a hard time compiling the old V9 version.

I am the original contributor of this tool.

### CODE REVIEW

**Comment by boredabdel** [comment]
I will review this one

**Comment by boredabdel** [comment]
First round of review. Left few comments.

Also please add the licensing header to all html and json files. These are considered source code.

File: tools/bq-visualizer/src/app/app.component.html
Comment (cbradgoog) [inline tools/bq-visualizer/src/app/app.component.html:7]: `<button>` has no `href` attribute, so I do not believe this works.

File: tools/bq-visualizer/src/app/bq_query_plan.ts
Comment (cbradgoog) [inline tools/bq-visualizer/src/app/bq_query_plan.ts:54]: nit: remove

File: tools/bq-visualizer/src/app/bq_query_plan.ts
Comment (cbradgoog) [inline tools/bq-visualizer/src/app/bq_query_plan.ts:223]: Please consider using `Array.prototype.slice` here.

File: tools/bq-visualizer/src/app/job/job.component.html
Comment (cbradgoog) [inline tools/bq-visualizer/src/app/job/job.component.html:37]: This `<p>` tag is not closed.

File: tools/bq-visualizer/src/app/projects/projects.component.ts
Comment (cbradgoog) [inline tools/bq-visualizer/src/app/projects/projects.component.ts:18]: nit: remove

**Review by cbradgoog — COMMENTED** [review]
Since I don’t know if you are coding against a certain coding standard, I omitted comments about code style, such as use of `new RegExp` vs. `/regex/` notation or consistency of `camelCasing`. Let me know if you would like to see those as well.

File: tools/bq-visualizer/src/app/bq_query_plan.ts
Comment (cbradgoog) [inline tools/bq-visualizer/src/app/bq_query_plan.ts:207]: I recommend creating an interface for this for better readability. For example:
```
export interface OnSelectHandler<T> {
  (chart: T, data: object): void;
}
```

Also this can be reused in both here and in `asSlotUsageChart` below.

File: tools/bq-visualizer/src/app/app.component.html
Comment (smeyn) [reply tools/bq-visualizer/src/app/app.component.html:7]: changed to an <a> element

**Comment by smeyn** [comment]
DId all, except adding licensing header to .json files:
- one is maintained by ng/npm so that would be overridden
- I did manage to add them to tsconfig.json as it appears to accept them
- with all the others it would flag them as an error

**Comment by boredabdel** [comment]
Thanks @smeyn 

@cbradgoog all is good on my side. If you have any more comments go ahead. Once i get your LGTM i will merge this one

**Review by cbradgoog — APPROVED** [review]
LGTM. Thanks for making those changes!

**Review by boredabdel — COMMENTED** [review]
LGTM

**Pull request merged**
